libr/crypto/p/crypto_des.c

```diff
@@ -0,0 +1,392 @@
+	   *deskeylo = ROTL28(*deskeylo, 1);
+	   *deskeyhi = ROTL28(*deskeyhi, 1);
+	} else {
+	   *deskeylo = ROTL28(*deskeylo, 2);
+	   *deskeyhi = ROTL28(*deskeyhi, 2);
+	}
+
+    deslo = *deskeylo;
+    deshi = *deskeyhi;
+
+	*keylo =((deslo << 4)  & 0x24000000) | ((deslo << 28) & 0x10000000) |
+			((deslo << 14) & 0x08000000) | ((deslo << 18) & 0x02080000) |
+			((deslo << 6)  & 0x01000000) | ((deslo << 9)  & 0x00200000) |
+			((deslo >> 1)  & 0x00100000) | ((deslo << 10) & 0x00040000) |
+			((deslo << 2)  & 0x00020000) | ((deslo >> 10) & 0x00010000) |
+			((deshi >> 13) & 0x00002000) | ((deshi >> 4)  & 0x00001000) |
+			((deshi << 6)  & 0x00000800) | ((deshi >> 1)  & 0x00000400) |
+			((deshi >> 14) & 0x00000200) | ((deshi)       & 0x00000100) |
+			((deshi >> 5)  & 0x00000020) | ((deshi >> 10) & 0x00000010) |
+			((deshi >> 3)  & 0x00000008) | ((deshi >> 18) & 0x00000004) |
+			((deshi >> 26) & 0x00000002) | ((deshi >> 24) & 0x00000001);
+
+	*keyhi =((deslo << 15) & 0x20000000) | ((deslo << 17) & 0x10000000) |
+			((deslo << 10) & 0x08000000) | ((deslo << 22) & 0x04000000) |
+			((deslo >> 2)  & 0x02000000) | ((deslo << 1)  & 0x01000000) |
+			((deslo << 16) & 0x00200000) | ((deslo << 11) & 0x00100000) |
+			((deslo << 3)  & 0x00080000) | ((deslo >> 6)  & 0x00040000) |
+			((deslo << 15) & 0x00020000) | ((deslo >> 4)  & 0x00010000) |
+			((deshi >> 2)  & 0x00002000) | ((deshi << 8)  & 0x00001000) |
+			((deshi >> 14) & 0x00000808) | ((deshi >> 9)  & 0x00000400) |
+			((deshi)       & 0x00000200) | ((deshi << 7)  & 0x00000100) |
+			((deshi >> 7)  & 0x00000020) | ((deshi >> 3)  & 0x00000011) |
+			((deshi << 2)  & 0x00000004) | ((deshi >> 21) & 0x00000002);
+}
+
+static void des_round(ut32 *buflo, ut32 *bufhi, ut32 *roundkeylo, ut32 *roundkeyhi) {
+	ut32 perm, lo, hi;
+	if (!buflo || !bufhi || !roundkeylo || !roundkeyhi) return;
+    lo = *buflo;
+    hi = *bufhi;
+	perm = hi ^ (*roundkeylo);
+	lo ^= sbox2[(perm >> 24) & 0x3F];
+	lo ^= sbox4[(perm >> 16) & 0x3F];
+	lo ^= sbox6[(perm >> 8) & 0x3F];
+	lo ^= sbox8[perm & 0x3F];
+	perm = ROTR(hi, 4) ^ (*roundkeyhi);
+	lo ^= sbox1[(perm >> 24) & 0x3F];
+	lo ^= sbox3[(perm >> 16) & 0x3F];
+	lo ^= sbox5[(perm >> 8) & 0x3F];
+	lo ^= sbox7[perm & 0x3F];
+	perm = hi;
+	*bufhi = lo;
+	*buflo = perm;
+}
+
+static int des_encrypt (struct des_state *st, const ut8 *input, ut8 *output) {
+    if (!st || !input || !output) return false;
+	st->buflo = be32(input + 0);
+	st->bufhi = be32(input + 4);
+ 
+	//first permutation
+	des_permute_block0(&st->buflo, &st->bufhi);
+
+ 	for (st->i = 0; st->i < 16; st->i++) {
+       des_round(&st->buflo, &st->bufhi, &st->keylo[st->i], &st->keyhi[st->i]);
+	}
+ 	//last permutation
+	des_permute_block1(&st->bufhi, &st->buflo);
+ 
+	//result
+    wbe32(output + 0, st->bufhi);
+    wbe32(output + 4, st->buflo);
+
+    return true;
+}
+
+static int des_decrypt (struct des_state *st, const ut8 *input, ut8 *output) {
+    if (!st || !input || !output) return false;

```

Comment:
add braces here and one space before every ( except in function signature definition